### PR TITLE
Refactor Proxy with ImplementFunc to allow override impl

### PR DESCRIPTION
**Pull request by yongjiapro**
**What this PR does**:

Ftr: Add Prox ImplementFunc to allow override impl 

重构Proxy，添加ImplementFunc函数，允许项目对Proxy的代理实现进行重新实现。

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

在网关泛调用场景下，当前GenericService实现需要修改对返回数据的处理逻辑。
目前ProxyFactory允许自定义注册，但其创建的proxy.Proxy无法重新实现。

```release-note
Ftr: Add Proxy ImplementFunc to allow override impl 
```

### CODE REVIEW

File: common/proxy/proxy.go [hunk 1]
```diff
@@ -40,23 +40,47 @@ import (
...
 	attachments map[string]string
-
-	once sync.Once
+	implement   ImplementFunc
+	once        sync.Once
 }
 
+type (
+	// ProxyOption a function to init Proxy with options
+	// nolint
```
Comment (gaoxinge) [inline common/proxy/proxy.go:51]: Remove `// nolint`.

File: common/proxy/proxy.go [hunk 2]
```diff
@@ -40,23 +40,47 @@ import (
...
 // NewProxy create service proxy.
-func NewProxy(invoke protocol.Invoker, callBack interface{}, attachments map[string]string) *Proxy {
-	return &Proxy{
+func NewProxy(invoke protocol.Invoker, callback interface{}, attachments map[string]string) *Proxy {
+	return NewProxyWith(invoke, callback, attachments,
+		WithProxyImplementFunc(DefaultProxyImplementFunc))
+}
+
+// NewProxy create service proxy.
+func NewProxyWith(invoke protocol.Invoker, callback interface{}, attachments map[string]string, opts ...ProxyOption) *Proxy {
```
Comment (gaoxinge) [inline common/proxy/proxy.go:68]: Rename `NewProxyWith` to `NewProxyWithOptions`.

File: common/proxy/proxy.go [hunk 3]
```diff
@@ -66,6 +90,29 @@ func NewProxy(invoke protocol.Invoker, callBack interface{}, attachments map[str
 //  		Yyy func(ctx context.Context, args []interface{}, rsp *Zzz) error
 // 		}
 func (p *Proxy) Implement(v common.RPCService) {
+	p.once.Do(func() {
+		p.implement(p, v)
+		p.rpc = v
+	})
+
```
Comment (gaoxinge) [inline common/proxy/proxy.go:97]: Remove blank.

**Comment by yongjiapro** [comment]
Rename and format code, Done

**Review by gaoxinge — APPROVED** [review]

**Review by zouyx — APPROVED** [review]

**Comment by AlexStocks** [comment]
pls change the target branch to 1.5

**Comment by yongjiapro** [comment]
> pls change the target branch to 1.5

Done

**Review by cityiron — APPROVED** [review]

**Comment by cityiron** [comment]
Tks, I will merge to 1.5 and add this for 1.5.6

**Pull request merged**